src/expr/impl/src/scalar/to_char.rs

```diff
@@ -138,3 +154,193 @@ fn timestamptz_to_char3(
     write!(writer, "{}", format).unwrap();
     Ok(())
 }
+
+#[function(
+    "to_char(interval, varchar) -> varchar",
+    prebuild = "ChronoPattern::compile($1, true)"
+)]
+fn interval_to_char(
+    interval: Interval,
+    pattern: &ChronoPattern,
+    writer: &mut impl Write,
+) -> Result<()> {
+    for iter in pattern.borrow_dependent() {
+        println!("iter: {:?}", iter);
+        format_inner(writer, interval, iter)?;
+    }
+    Ok(())
+}
+
+fn adjust_to_iso_year(interval: Interval) -> Result<i32> {
+    let start = risingwave_common::types::Timestamp(
+        NaiveDate::from_ymd_opt(0, 1, 1)
+            .unwrap()
+            .and_hms_opt(0, 0, 0)
+            .unwrap(),
+    );
+    let interval = Interval::from_month_day_usec(interval.months(), interval.days(), 0);
+    let date = timestamp_interval_add(start, interval)?;
+    Ok(date.0.iso_week().year())
+}
+
+fn format_inner(w: &mut impl Write, interval: Interval, item: &Item<'_>) -> Result<()> {
+    match *item {
+        Item::Literal(s) | Item::Space(s) => {
+            w.write_str(s).unwrap();
+            Ok(())
+        }
+        Item::OwnedLiteral(ref s) | Item::OwnedSpace(ref s) => {
+            w.write_str(s).unwrap();
+            Ok(())
+        }
+        Item::Numeric(ref spec, _) => {
+            use chrono::format::Numeric::*;
+            match *spec {
+                Year => {
+                    let year = interval.years_field();
+                    if year < 0 {
+                        write!(w, "{:+01$}", year, 5).unwrap()
```

Comment:
Today I learned the "1$" format. But maybe embedding looks simpler? 
```suggestion
                        write!(w, "{:+05}", year).unwrap()
```

Reply:
Updated, Thx!